Error: The commit wasn't found on the repo

I have a Coveralls setup that works perfectly in our public repos, but I used the same setup in a private repo, obviously adding the Coveralls token to Travis, and now I get this weird error about the commit not being found. Any ideas on what to do here? The PHPUnit coverage is reporting correctly to Coveralls but the JS coverage is failing. I've commented it out temporarily until I can find a solution, but that's not ideal.

```bash
> wp-scripts test-unit-js --coverage --coverageReporters=text-lcov | coveralls
TravisFoldPassesReporter
...1 passing test.
/home/travis/build/xwp/material-theme-builder-wp/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 422 {"message":"Travis API Error: #\u003cCoveralls::Mash message=\"The job(285764075) couldn't be found\"\u003e. The commit  wasn't found on the repo xwp/material-theme-builder-wp","error":true}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! material-theme-builder-wp@ test:js:coveralls: `wp-scripts test-unit-js --coverage --coverageReporters=text-lcov | coveralls`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the material-theme-builder-wp@ test:js:coveralls script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-02-11T07_08_20_775Z-debug.log
```

--- a/lib/getOptions.js
+++ b/lib/getOptions.js
@@ -26,7 +26,7 @@ const getBaseOptions = cb => {
     options.service_name = 'travis-ci';
     options.service_job_id = process.env.TRAVIS_JOB_ID;
     options.service_pull_request = process.env.TRAVIS_PULL_REQUEST;
-    git_commit = 'HEAD';
+    git_commit = process.env.TRAVIS_COMMIT || 'HEAD';
     git_branch = process.env.TRAVIS_BRANCH;
   }
 

--- a/lib/getOptions.js
+++ b/lib/getOptions.js
@@ -163,7 +163,9 @@ const getBaseOptions = cb => {
 
   // try to get repo token and service name from .coveralls.yml file
   if (coveralls_yaml_conf) {
-    options.repo_token = coveralls_yaml_conf.repo_token;
+    if (coveralls_yaml_conf.repo_token) {
+      options.repo_token = coveralls_yaml_conf.repo_token;
+    }
     if (coveralls_yaml_conf.service_name) {
       options.service_name = coveralls_yaml_conf.service_name;
     }

--- a/lib/getOptions.js
+++ b/lib/getOptions.js
@@ -132,9 +132,6 @@ const getBaseOptions = cb => {
   }
 
   options.run_at = process.env.COVERALLS_RUN_AT || JSON.stringify(new Date()).slice(1, -1);
-  if (process.env.COVERALLS_SERVICE_NAME) {
-    options.service_name = process.env.COVERALLS_SERVICE_NAME;
-  }
 
   if (process.env.COVERALLS_SERVICE_JOB_ID) {
     options.service_job_id = process.env.COVERALLS_SERVICE_JOB_ID;
@@ -152,24 +149,39 @@ const getBaseOptions = cb => {
     options.parallel = true;
   }
 
-  // try to get the repo token as an environment variable
-  if (process.env.COVERALLS_REPO_TOKEN) {
-    options.repo_token = process.env.COVERALLS_REPO_TOKEN;
-  } else {
-    // try to get the repo token from a .coveralls.yml file
+  // load a .coveralls.yml file
+  const coveralls_yaml_conf = (() => {
     const yml = path.join(process.cwd(), '.coveralls.yml');
     try {
       if (fs.statSync(yml).isFile()) {
-        const coveralls_yaml_conf = yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
-        options.repo_token = coveralls_yaml_conf.repo_token;
-        if (coveralls_yaml_conf.service_name) {
-          options.service_name = coveralls_yaml_conf.service_name;
-        }
+        return yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
       }
     } catch (_) {
-      logger.warn('Repo token could not be determined.  Continuing without it. ' +
-                  'This is necessary for private repos only, so may not be an issue at all.');
+      logger.debug('No valid .coveralls.yml file found');
     }
+  })();
+
+  // try to get repo token and service name from .coveralls.yml file
+  if (coveralls_yaml_conf) {
+    options.repo_token = coveralls_yaml_conf.repo_token;
+    if (coveralls_yaml_conf.service_name) {
+      options.service_name = coveralls_yaml_conf.service_name;
+    }
+  }
+
+  // try to get the repo token as an environment variable
+  if (process.env.COVERALLS_REPO_TOKEN) {
+    options.repo_token = process.env.COVERALLS_REPO_TOKEN;
+  }
+
+  if (!options.repo_token) {
+    logger.warn('Repo token could not be determined.  Continuing without it. ' +
+                'This is necessary for private repos only, so may not be an issue at all.');
+  }
+
+  // try to get the service name as an environment variable
+  if (process.env.COVERALLS_SERVICE_NAME) {
+    options.service_name = process.env.COVERALLS_SERVICE_NAME;
   }
 
   if (process.env.COVERALLS_FLAG_NAME) {

--- a/lib/getOptions.js
+++ b/lib/getOptions.js
@@ -176,7 +176,7 @@ const getBaseOptions = cb => {
     options.repo_token = process.env.COVERALLS_REPO_TOKEN;
   }
 
-  if (!options.repo_token) {
+  if ('travis-pro' === options.service_name && !options.repo_token) {
     logger.warn('Repo token could not be determined.  Continuing without it. ' +
                 'This is necessary for private repos only, so may not be an issue at all.');
   }

--- a/test/getOptions.js
+++ b/test/getOptions.js
@@ -45,6 +45,9 @@ describe('getBaseOptions', () => {
   it('should set service_name and service_job_id if it\'s running on travis-ci', done => {
     testTravisCi(getBaseOptions, done);
   });
+  it('should set service_name and service_job_id if it\'s running on travis-pro', done => {
+    testTravisPro(getBaseOptions, done);
+  });
   it('should set service_name and service_job_id if it\'s running on jenkins', done => {
     testJenkins(getBaseOptions, done);
   });
@@ -136,6 +139,9 @@ describe('getOptions', () => {
   });
   it('should set service_name and service_job_id if it\'s running on travis-ci', done => {
     testTravisCi(getOptions, done);
+  });
+  it('should set service_name and service_job_id if it\'s running on travis-pro', done => {
+    testTravisPro(getOptions, done);
   });
   it('should set service_name and service_job_id if it\'s running on jenkins', done => {
     testJenkins(getOptions, done);
@@ -343,6 +349,23 @@ const testTravisCi = (sut, done) => {
     options.service_name.should.equal('travis-ci');
     options.service_job_id.should.equal('1234');
     options.service_pull_request.should.equal('123');
+    done();
+  });
+};
+
+const testTravisPro = (sut, done) => {
+  const file = path.join(process.cwd(), '.coveralls.yml');
+  const service_name = 'travis-pro';
+  fs.writeFileSync(file, `service_name: ${service_name}`);
+  process.env.TRAVIS = 'TRUE';
+  process.env.TRAVIS_JOB_ID = '1234';
+  process.env.TRAVIS_COMMIT = 'a12s2d3df4f435g45g45g67h5g6';
+  sut((err, options) => {
+    should.not.exist(err);
+    options.service_name.should.equal(service_name);
+    options.service_job_id.should.equal('1234');
+    options.git.head.id.should.equal('a12s2d3df4f435g45g45g67h5g6');
+    fs.unlinkSync(file);
     done();
   });
 };

